Source/CarthageKit/Project.swift

```diff
@@ -453,23 +495,22 @@ public final class Project {
 						.flatMap(.concat, transform: unzip(archive:))
 						.flatMap(.concat) { directoryURL in
 							return frameworksInDirectory(directoryURL)
+								.flatMap(.merge, transform: self.matchingSwiftVersionURL)
 								.flatMap(.merge, transform: self.copyFrameworkToBuildFolder)
 								.flatMap(.merge) { frameworkURL in
 									return self.copyDSYMToBuildFolderForFramework(frameworkURL, fromDirectoryURL: directoryURL)
 										.then(self.copyBCSymbolMapsToBuildFolderForFramework(frameworkURL, fromDirectoryURL: directoryURL))
 								}
 								.on(completed: {
 									_ = try? FileManager.default.trashItem(at: checkoutDirectoryURL, resultingItemURL: nil)
+								}, disposed: {
+									_ = try? FileManager.default.removeItem(at: directoryURL)
 								})
-								.then(SignalProducer<URL, CarthageError>(value: directoryURL))
 						}
-						.attemptMap { (temporaryDirectoryURL: URL) -> Result<Bool, CarthageError> in
-							do {
-								try FileManager.default.removeItem(at: temporaryDirectoryURL)
-								return .success(true)
-							} catch let error as NSError {
-								return .failure(.writeFailed(temporaryDirectoryURL, error))
-							}
+						.map { _ in true }
+						.flatMapError { error in
+							self._projectEventsObserver.send(value: .skippedInstallingBinaries(project, error.description))
```

Comment:
I followed the pattern on `skippedDownloadingBinaries` here. I do think it would be helpful to include the project information as well, but I suppose I could fold that in to an error.